src/components/annotations/draw.js

```diff
@@ -562,19 +610,22 @@ function drawRaw(gd, options, index, subplotId, xa, ya) {
                         var ycenter = annxy0[1] + dy;
                         annTextGroupInner.call(Drawing.setTranslate, xcenter, ycenter);
 
-                        modifyItem('x', xa ?
-                            xa.p2r(xa.r2p(options.x) + dx) :
-                            (options.x + (dx / gs.w)));
-                        modifyItem('y', ya ?
-                            ya.p2r(ya.r2p(options.y) + dy) :
-                            (options.y - (dy / gs.h)));
+                        modifyItem('x',
+                            shiftPosition(xa, options.x, dx, gs.w, false, options.xref));
```

Comment:
We could possibly refactor `shiftPosition` function in a way here & in other places we simply call it like the following:
```js
shiftPosition(xa, dx, gs, options);
```

Reply:
In order to achieve that the `shiftPosition` arguments may look like this:
```js
function shiftPosition(axa, dAx, gs, opts) {
    var axLetter = axa._id.charAt(0);
    var optAx = opts[axLetter];
    var axRef = opts[axLetter + 'ref'];
    var vertical = axLetter === 'y';
    var gsDim = vertical ? gs.h : gs.w;
```

Reply:
fixed by 6eba309ac6a535bdf7849c070be56e8f5ac4ec07

Reply:
Resolved thanks to https://github.com/plotly/plotly.js/pull/5014/commits/6eba309ac6a535bdf7849c070be56e8f5ac4ec07 and https://github.com/plotly/plotly.js/pull/5014/commits/759d37bf9e1eac56cef5d1e4df44e899e20ffadc.